variants/github_actions_ci/workflows/deploy_to_ec2.yml

```diff
@@ -0,0 +1,78 @@
+name: deploy_to_ec2
+on:
+  workflow_call:
+    inputs:
+      environment:
+        type: string
+        required: true
+      assume_role_arn:
+        type: string
+        required: true
+      aws_region:
+        type: string
+        required: true
+      timeout_minutes:
+        type: number
+        required: false
+        default: 15
+    secrets:
+      ssh_private_key:
+        required: true
+      slack_webhook:
+        required: true
+jobs:
+  deploy_to_ec2_via_oidc:
+    name: Deploy to AWS EC2 with Capistrano, authenticated by Github OIDC
+    runs-on: ubuntu-latest
+    timeout-minutes: ${{ inputs.timeout_minutes }}
+    permissions:
+      # These permissions are needed to interact with GitHub's OIDC Token
+      # endpoint.
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+      - name: Configure AWS credentials using OpenID Connect to AssumeRole
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-session-name: GitHubActionsDeployment
```

Comment:
🐘 I'd prefer to omit this since we're not customizing it, to avoid tempting people to do that (I'd say we could use `env.GITHUB_REPOSITORY`, but I don't know what characters we're allowed to use for this)